core/src/main/java/org/apache/calcite/rel/externalize/RelJsonReader.java

```diff
@@ -199,6 +199,9 @@ public boolean getBoolean(String tag, boolean default_) {
       public List<RexNode> getExpressionList(String tag) {
         @SuppressWarnings("unchecked")
         final List<Object> jsonNodes = (List) jsonRel.get(tag);
+        if (jsonNodes == null) {
+          return null;
```

Comment:
When the `jsonNodes` could be null ?

Reply:
when serialize TableModify, the 'updateColumnList' and 'sourceExpressionList' may be null. 
For exampl, for this relnode tree
```
LogicalTableModify(table=[[scott, EMP]], operation=[INSERT], flattened=[false])
  LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7])
    LogicalTableScan(table=[[scott, EMP]])
```
the serialized json string is
```
{
  "rels": [
    {
      "id": "0",
      "relOp": "LogicalTableScan",
      "table": [
        "scott",
        "EMP"
      ],
      "inputs": []
    },
    {
      "id": "1",
      "relOp": "LogicalProject",
      "fields": [
        "EMPNO",
        "ENAME",
        "JOB",
        "MGR",
        "HIREDATE",
        "SAL",
        "COMM",
        "DEPTNO"
      ],
      "exprs": [
        {
          "input": 0,
          "name": "$0"
        },
        {
          "input": 1,
          "name": "$1"
        },
        {
          "input": 2,
          "name": "$2"
        },
        {
          "input": 3,
          "name": "$3"
        },
        {
          "input": 4,
          "name": "$4"
        },
        {
          "input": 5,
          "name": "$5"
        },
        {
          "input": 6,
          "name": "$6"
        },
        {
          "input": 7,
          "name": "$7"
        }
      ]
    },
    {
      "id": "2",
      "relOp": "LogicalTableModify",
      "table": [
        "scott",
        "EMP"
      ],
      "operation": "INSERT",
      "flattened": false
    }
  ]
}
```
When trying to get updateColumnList by `input.getExpressionList("sourceExpressionList")`, the jsonNodes will be null.

Reply:
Well, the `targetColumnNameList` is always null for operation `INSERT` and `DELETE`;
the `sourceExpressionList` is always null for `INSERT`, `DELETE` and `MERGE`.

Reply:
yes, I update the null check logic in constructor.